File: 0118-pascals-triangle/solution.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        if(numRows <= 1) return {{numRows}};
        vector<vector<int>> dp(numRows);
        dp[0] = {1};
        dp[1] = {1,1};

        for(int j = 2; j < numRows; j++){
            auto& t = dp[j -1];

            vector<int> row(t.size()+1, 1);

            for(int i = 1; i < row.size() - 1; i++){
                row[i] = t[i] + t[i-1];
            }
            dp[j] = row;
        }

        return dp;
    
        


    }
};

File: 0118-pascals-triangle/solution.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 1:
            return [[1]]
            
        dp = [[1], [1,1]]

        for i in range(1, numRows - 1):
            curr = [1] * (len(dp[i]) + 1)
            k = 0
            for j in range(1, len(curr) - 1):
                curr[j] = dp[i][k] + dp[i][k + 1]
                k+=1

            dp.append(curr)

        return dp
            
